packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1ArbitrumGateway.sol

```diff
@@ -220,7 +252,7 @@ abstract contract L1ArbitrumGateway is L1ArbitrumMessenger, TokenGateway {
             res = getOutboundCalldata(_l1Token, _from, _to, _amount, extraData);
 
             seqNum = createOutboundTx(
-                _from,
+                _refundTo,
```

Comment:
this will mess up the event that is emitted.
We could instead add the "refund" address field separate from "user" to [L1ArbitrumMessenger](https://github.com/OffchainLabs/arbitrum/blob/c8ab7a69ce1176aaca551607b8dbc1153d31581f/packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/L1ArbitrumMessenger.sol#L74-L75) that is used.

This might create stack size issues, which you can get around by using structs

Reply:
resolved in commit 0f1c0b3637bb8d4dc4de4adf9742060459258fe7

Reply:
stack size issue is avoided by reordering `_refundTo` before `_user` since that is the order we will consume in L1ArbitrumMessenger